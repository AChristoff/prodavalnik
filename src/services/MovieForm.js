import React, { Component } from 'react'

export default class MovieForm extends Component {
  state = {
    text: '',
  }

  render() {
    const {submitForm} = this.props;
    const {text}= this.state;


    return (
      <form 
        data-testid="movie-form" 
        onSubmit={
          () => submitForm({
            text
          })
      }>
        <input type="text"/>
        <button type="submit">Submit</button>
      </form>
    )
  }
}
