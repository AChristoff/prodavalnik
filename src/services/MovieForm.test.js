import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import MovieForm from './MovieForm';

afterEach(cleanup);

const onSubmit = jest.fn();

test('<MovieForm>', () => {
  const { debug, getByTestId, queryByTestId, container, getByText } = render(<MovieForm submitForm={onSubmit}/>);

  expect(queryByTestId('movie-form')).toBeTruthy();

  fireEvent.click(getByText('Submit'));

  expect(onSubmit).toHaveBeenCalledTimes(1);
  // debug();
})