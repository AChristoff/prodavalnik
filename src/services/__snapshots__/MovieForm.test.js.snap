// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie> 1`] = `
<div>
  <h2
    data-testid="page-title"
  >
    New Movie
  </h2>
  <form
    data-testid="movie-form"
  >
    <input
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
